### PR TITLE
Add the plugin development documentation

### PR DESCRIPTION
**Pull request by quantumm**
This is a tutorial for the plugin development based on the example of the [PI E870 plugin](https://github.com/PyMoDAQ/pymodaq_plugins_physik_instrumente/pull/4).

### CODE REVIEW

**Comment by quantumm** [comment]
It has been sent to the v3 branch because there is probably some minor changes to make it v4 compatible.

**Comment by quantumm** [comment]
Have a good reading ;)

**Comment by seb5g** [comment]
That's quite a bit of a book!! It will be a very valuable piece of documentation for beginners.

I don't have much comment at the moment except a few below:

1. The first concerns the vocabulary used concerning pymodaq's plugin. Initially they were only for instruments, but as it turns out they are now also used for pid models, extensions, extending functionalities (through factory patterns)... That is why I introduced a dedicated vocabulary concerning those:

image

I think therefore we could speak about a pymodaq plugin as a package giving new functionalities to pymodaq and refer to the type of functionalities we are talking about:

- instrument 
- extensions
- pid models
- factories (not ready yet)

a plugin with a manufacturer name will be considered as an *instrument plugin* but one could also consider a plugin named:
*pymodaq_plugins_attolab* that contains your specific pid models and factories (for instance specific exports from hdf5 only you are using)
or a plugin called:
*pymodaq_plugins_console* that enables a ipython console from the dahboard...

Do you think you could talk about this in your tutorial? 

2. Concerning the third party libraries (pylablib, instrumental-lib, ...)
You should maybe tell what they are good at. You already said quite a bit about pylablib so maybe you should say that instrumental-lib is particularly good to create drivers from dll written from C where one have the header file. For autoprocessing of the functions signature... And I like you to mention pymeasure I'm being more involved with. This libray is very efficient for all instruments that communicate through ascii commands (pyvisa basically) and make drivers very easy to use and develop. I just produced one for the elliptec thorlabs piezo drivers in a day!! I'm also going to invite those guys to the pymodaq's days!

3. Maybe say that your example is one dealing with C dlls but that a big zoology exists (ascci commands, C/C++ dlls (using instrumental-lib or ctypes or manufacturer wrappers) but also .net libraries using pythonnet...

Thanks a lot for this work, we will publish it in the v3 after you modified your text according to what i just wrote and then port it to the v4 as is. In this v4 (pymodaq_dev branch actually), the documentation layout is quite different and I'll make a specific tutorial entry.

**Comment by quantumm** [comment]
I think I have implemented all the corrections you asked.

Let me know if this is ok ;)

File: docs/src/usage/plugin_development.rst [hunk 1]
```diff
@@ -167,7 +174,7 @@ First, we have to install PyMoDAQ in a dedicated Python environment, that we wil
 
 __ https://pymodaq.cnrs.fr/en/pymodaq-dev/user_folder/installation.html#
 
-Let's try to install it with the plugin manager. In your terminal, within your activated environment where you just installed PyMoDAQ (the lign of your terminal should start with *(pmd_dev)*), type
+Now that PyMoDAQ is installed and you have activated your environment (the lign of your terminal should start with *(pmd_dev)*), we will try to install the PI instrument plugin with the plugin manager. In your terminal, execute the following command
```
Comment (seb5g) [inline docs/src/usage/plugin_development.rst:183]: replace lign by line

**Review by seb5g — APPROVED** [review]

**Review by seb5g — CHANGES_REQUESTED** [review]
please change the bits about the label/reference

File: docs/src/usage/plugin_development.rst [hunk 2]
```diff
@@ -1,15 +1,16 @@
...
-=============================
+Story of an instrument plugin development
+=========================================
 
-In this section we will explain step by step the way to develop a plugin, so that you can control your specific device through PyMoDAQ. If you are not familiar with it, start with this `introduction to plugins`__.
+In this tutorial, we will explain step by step the way to develop an `instrument plugin`__. It is a specific type of `plugin`__, that will allow you to control your device through PyMoDAQ. In the following, we may use the term *plugin* in order to ease the reading of the tutorial, but it will always actually refer to an *instrument plugin*.
 
-__ https://pymodaq.cnrs.fr/en/pymodaq-dev/usage/modules/Plugins.html
+__ https://pymodaq.cnrs.fr/en/latest/developer_folder/instrument_plugins.html
+__ https://pymodaq.cnrs.fr/en/latest/glossary.html#term-Plugin
```
[the remark this answers is not in the document]
Comment (quantumm) [reply docs/src/usage/plugin_development.rst:7]: Done in f05aaadf70ef0be4245c76178e70ae6d91461f4e.

**Review by seb5g — APPROVED** [review]

**Pull request merged**